pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -1,50 +1,63 @@
-{ copyDesktopItems, fetchurl, jre, lib, makeDesktopItem, makeWrapper, stdenv, udev, xorg }:
+{ copyDesktopItems
+, fetchFromGitHub
+, fetchurl
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenv
 
-stdenv.mkDerivation (finalAttrs: {
+, # dependencies
+  jre
+, udev
+, xorg
+}:
+
+let
   pname = "atlauncher";
-  version = "3.4.35.4";
+  version = "3.4.35.5";
 
-  src = fetchurl {
-    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${finalAttrs.version}/ATLauncher-${finalAttrs.version}.jar";
-    hash = "sha256-M8ygN70yizJM6VEffBh/lH/DneKAzQ5UFzc3g51dja0=";
+  srcs = {
+    source = (fetchFromGitHub {
+      owner = "ATLauncher";
+      repo = "ATLauncher";
+      rev = "v${version}";
+      sha256 = "sha256-9jOvElwb3jmO18MZOFqbB/mF75SdKufZvM1oWLKZDII=";
+    });
+    jar = (fetchurl {
+      url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${version}/ATLauncher-${version}.jar";
+      hash = "sha256-Y3Xe85Pte55ijIpJubOCfqIodSfpigJte2i/6eYzyO8=";
+    });
   };
 
-  env.ICON = fetchurl {
-    url = "https://atlauncher.com/assets/images/logo.svg";
-    hash = "sha256-XoqpsgLmkpa2SdjZvPkgg6BUJulIBIeu6mBsJJCixfo=";
-  };
+  src-meta-dir = "${srcs.source}/packaging/linux/_common";
+in
+stdenv.mkDerivation {
+  inherit pname version;
 
+  src = srcs.jar;
   dontUnpack = true;
 
   nativeBuildInputs = [ copyDesktopItems makeWrapper ];
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin $out/share/java
-    cp $src $out/share/java/ATLauncher.jar
+    install -D -m 444 $src $out/share/java/ATLauncher.jar
 
     makeWrapper ${jre}/bin/java $out/bin/atlauncher \
       --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ xorg.libXxf86vm udev ]}" \
       --add-flags "-jar $out/share/java/ATLauncher.jar" \
       --add-flags "--working-dir \"\''${XDG_DATA_HOME:-\$HOME/.local/share}/ATLauncher\"" \
       --add-flags "--no-launcher-update"
 
-    mkdir -p $out/share/icons/hicolor/scalable/apps
-    cp $ICON $out/share/icons/hicolor/scalable/apps/atlauncher.svg
-
     runHook postInstall
   '';
 
-  desktopItems = [
-    (makeDesktopItem {
-      categories = [ "Game" ];
-      desktopName = "ATLauncher";
-      exec = "atlauncher";
-      icon = "atlauncher";
-      name = "atlauncher";
-    })
-  ];
+  postInstall = ''
+    install -D -m 644 ${src-meta-dir}/atlauncher.svg $out/share/icons/hicolor/scalable/apps/atlauncher.svg
```

Comment:
```suggestion
    install -Dm644 ${packagingDir}/atlauncher.svg $out/share/icons/hicolor/scalable/apps/atlauncher.svg
```

Reply:
Merging the flags into one harms readability, IMO.